components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java

```diff
@@ -9759,6 +9759,22 @@ public static String getStoreUrl() throws APIManagementException {
                 .getAPIManagerConfiguration().getApiGatewayEnvironments();
     }
 
+    /**
+     * Get default (first) vhost of the given environment
+     * @param environmentName name of the environment
+     * @return default vhost of environment
+     */
+    public static VHost getDefaultVhostOfEnvironment(String environmentName) throws APIManagementException {
+        Map<String, Environment> environments = getEnvironments();
+        if (environments.get(environmentName) == null) {
+            throw new APIManagementException("Configured environment not found: " + environmentName);
```

Comment:
throwing such exception will break the import. Current implementation is done in a way that if there is a problem in the deployment environment, that environment will be skipped without deployment. 

@uvindra @pubudu538 @wasuradananjith shall we preserve the same experience?

Reply:
Since we are checking `if (gatewayEnvironmentsSet.contains(deploymentName))` before calling this, this exception will not be thrown, but the next exception in line 9773 will thrown, if not vhosts are configured in the environment

Reply:
We will proceed with your way. If the vhost entered is faulty, import should abort. not skip that environment.

Reply:
I think we can also remove calling `APIUtil. getDefaultVhostOfEnvironment()` since it makes another DB query, we can use existing already queried environments, and so I will remove this added util function. And will throw an exception when no vhosts are defined in the gateway environment